pkg/kubelet/gpu/nvidia/nvidia_gpu_manager.go

```diff
@@ -98,11 +105,15 @@ func (ngm *nvidiaGPUManager) Start() error {
 	if !os.IsNotExist(err) {
 		ngm.defaultDevices = append(ngm.defaultDevices, nvidiaUVMToolsDevice)
 	}
-
-	if err := ngm.discoverGPUs(); err != nil {
-		return err
-	}
-
+	// retry discoverGPUs indefinitely in case drivers have not been installed
+	go func() {
+		for {
+			if found := ngm.discoverGPUs(devicesProcFile, devDirectory); found == nil {
+				return
```

Comment:
We should log that GPU discover finished, so that there's an easy way to find out whether this goroutine is running or finished.